ComputerVision.cs

```diff
@@ -0,0 +1,518 @@
+            {
+                case ServiceMode.ObjectDetection:
+                    return objectsDetected == null ? false : true;
+                case ServiceMode.FaceDetection:
+                    return faceDetectionRects == null ? false : true;
+                default:
+                    return false;
+            }
+        }
+    }
+
+    /// <summary>
+    /// For more info about the computer vision server and the url suffix check MobiledgeX docs
+    /// for Rest https://developers.mobiledgex.com/guides-and-tutorials/computer-vision/how-to-computer-vision-api
+    /// for WebSockets https://mobiledgex.com/blog/2020/04/22/computer-vision-websocket-support
+    /// </summary>
+    string urlSuffix
+    {
+        get
+        {
+            switch (serviceMode)
+            {
+                case ServiceMode.ObjectDetection:
+                    return "/object/detect/";
+                case ServiceMode.FaceDetection:
+                    return "/detector/detect/";
+                default:
+                    return "";
+            }
+        }
+    }
+
+    /// <summary>
+    /// The complete url retreived from FindCloudlet+ url suffix
+    /// </summary>
+    static string url;
+    static bool urlFound; // trigger indicating RegisterAndFindCloudlet, GetAppPort() & GetUrl() occured
+    bool webRequestsLock = false; // to control the flow of sending web requests
+    bool wsStarted; // trigger indicating wether websocket connection have been intialized or not
+
+
+
+    #region Monobehaviour callbacks
+    IEnumerator Start()
+    {
+#if UNITY_EDITOR
+        GetEdgeConnection();
+        yield break;
+#endif
+        if (Application.internetReachability == NetworkReachability.NotReachable)
+        {
+            // Handle failure due no internet connection
+            throw new Exception("No Internet Connection");
+        }
+        else 
+        {
+            // MobiledgeX uses location for finding the closest cloudlet to the user
+            yield return StartCoroutine(MobiledgeX.LocationService.EnsureLocation());
+            GetEdgeConnection();
+        }
+    }
+
+    void OnGUI()
+    {
+        // show Detection Rects after the screen shot have been taken and the server have responded
+        if (showGUI)
+        {
+            DrawRectangles();
+        }
+    }
+
+    private void Update()
+    {
+        if (client == null)
+        {
+            return;
+        }
+        var cqueue = client.receiveQueue;
+        string msg;
+        while (cqueue.TryPeek(out msg))
+        {
+            cqueue.TryDequeue(out msg);
+            HandleServerRespone(msg);
+        }
+
+    }
+    #endregion
+
+    async Task GetEdgeConnection()
+    {
+        integration = new MobiledgeXIntegration();
+        integration.UseWifiOnly(true);
+        
+        try
+        {
+            bool cloudletFound = await integration.RegisterAndFindCloudlet("eu-mexdemo.dme.mobiledgex.net", 38001);
```

Comment:
hardcode?

Reply:
As discussed in a different PR, In the next release we will have the SDK regions in the SDK and the CV component will pull from those values.
However we are connecting only to EU region since in this region is the only one who have GPU enabled Cloudlets.